### PR TITLE
Added fetch self

### PR DESCRIPTION
**Pull request by AusCyberman**
you can now do !fetch self wm
to fetch yourself, could be reword to just be s or something similar

### CODE REVIEW

**Comment by 6gk** [comment]
lol nice
i was literally thinking of a similar feature a few hours ago

**Comment by AusCyberman** [comment]
> lol nice i was literally thinking of a similar feature a few hours ago

should it be `!fetch s `or self?

**Comment by 6gk** [comment]
Honestly I was imagining a direct `!fetch de/wm`, and if you're trying
to fetch someone with a name like that fuck em lol

**Comment by AusCyberman** [comment]
help, its so fucked, wtf is going on i just pulled from master

**Comment by AusCyberman** [comment]
> Honestly I was imagining a direct `!fetch de/wm`, and if you're trying to fetch someone with a name like that fuck em lol (obviously i'm not saying to do that, or even that it's a good idea, but that's just how i imagined it) I have no preference towards s or self or anything tbh

Done it

File: src/commands/fetch/fetch.rs [hunk 1]
```diff
@@ -10,14 +10,28 @@ pub async fn fetch(ctx: &client::Context, msg: &Message, mut args: Args) -> Comm
...
+            }
+        }
         Err(_) => msg.author.id,
     };
 
-    let desired_field = args.single_quoted::<String>().ok();
+    let desired_field = match desired_field {
+        None => args.single_quoted::<String>().ok(),
+        field => field,
+    };
```
Comment (elkowar) [inline src/commands/fetch/fetch.rs:34]: These could be combined into one match that returns a tuple, like
```
let (mentioned_user_id, desired_field) = match .....
```
thus cleaning this up a bit

**Review by elkowar — CHANGES_REQUESTED** [review]

File: src/commands/fetch/fetch.rs (hunk 1 above)
Comment (AusCyberman) [reply src/commands/fetch/fetch.rs]: all good

File: src/commands/fetch/fetch.rs [hunk 2]
```diff
@@ -10,15 +10,25 @@ pub async fn fetch(ctx: &client::Context, msg: &Message, mut args: Args) -> Comm
...
-        Ok(mentioned_user) => disambiguate_user_mention(&ctx, &guild, msg, &mentioned_user)
-            .await?
-            .ok_or(UserErr::MentionedUserNotFound)?,
-        Err(_) => msg.author.id,
+    let (desired_field, mentioned_user_id) = match args.single_quoted::<String>() {
+        Ok(mentioned_user) => {
+            if find_fetch_key_matching(&mentioned_user).is_some()
+                || IMAGE_KEY == mentioned_user.to_lowercase()
+            {
+                println!("does contain");
```
Comment (elkowar) [inline src/commands/fetch/fetch.rs:18]: this log statement should also be yote

**Pull request merged**